GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1693,12 +1693,12 @@ public struct DrawRefArgs
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-                        var gravatar = _avatarCache.GetImage($"{revision.AuthorEmail}.png", null);
+                        var gravatar = _gravatarService.GetImage(revision.AuthorEmail, null);
```

Comment:
I think that IImageCache should operate on Email instead of imageFileName.
Each cache implementation can store images differently, so it should be the DirectoryImageCache responsibility to generate file name from a given email.